themes/10up-block-theme/includes/blocks.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Blocks setup, site hooks and filters.
+ *
+ * @package TenupBlockTheme
+ */
+
+namespace TenupBlockTheme\Blocks;
+
+/**
+ * Set up theme defaults and register supported WordPress features.
+ *
+ * @return void
+ */
+function setup() {
+	add_action( 'init', 'TenupBlockTheme\Blocks\register_theme_blocks', 10, 0 );
+	add_action( 'init', 'TenupBlockTheme\Blocks\enqueue_theme_block_styles', 10, 0 );
+}
+
+/**
+ * Automatically registers all blocks that are located within the includes/blocks directory
+ *
+ * @return void
+ */
+function register_theme_blocks() {
+	// Register all the blocks in the theme.
+	if ( file_exists( TENUP_BLOCK_THEME_BLOCK_DIST_DIR ) ) {
+		$block_json_files = glob( TENUP_BLOCK_THEME_BLOCK_DIST_DIR . '*/block.json' );
+
+		foreach ( $block_json_files as $filename ) {
+			$block_folder = dirname( $filename );
+			$block        = register_block_type_from_metadata( $block_folder );
+
+			add_filter(
```

Comment:
**request:** can this go outside of the foreach loop? here we are calling add_filter for every block registered rather than once with all the blocks to add